src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CaseReferenceValidator.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd;
+
+import com.google.common.base.Strings;
+import io.vavr.control.Validation;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import java.util.Optional;
+import javax.annotation.Nonnull;
+
+import static io.vavr.control.Validation.invalid;
+import static java.lang.String.format;
+
+class CaseReferenceValidator {
+    private static final String ATTACH_TO_CASE_REFERENCE = "attachToCaseReference";
+
+    @Nonnull
+    Validation<String, String> validate(CaseDetails theCase) {
+        return Optional.ofNullable(theCase)
+            .map(CaseDetails::getData)
+            .map(data -> data.get(ATTACH_TO_CASE_REFERENCE))
+            .flatMap(this::validateCaseRef)
+            .orElseGet(() -> invalid("No case reference supplied"));
+    }
+
+    private Optional<Validation<String, String>> validateCaseRef(Object caseRef) {
+        Optional<Validation<String, String>> valid = Optional.of(caseRef)
```

Comment:
We only check if caseRef is a String having a digit in it. I think if we used conditions, it would be the clearest. Again, a suggestion only.

Reply:
The problem is that every part can be null or the wrong type so writing solid code means you need to check them.
Hence the functional chain.